### PR TITLE
merge & dash board & command box & autocomplete

### PR DESCRIPTION
**Pull request by ttochi**
- merge & separate
- dash board for new tab
- auto complete
- command box injection

### CODE REVIEW

File: src/html/dashboard.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+</div>
+
+<div id="right-side">
+  <button id="separate">SEPARATE</button>
+  <div id="tab-list">
+    <div class="tab">
+      <div class="delete"></div>
+      <div class="title"></div>
+      <div class="url"></div>
+      <!--<img class="screen" src="../../resources/icon.png" height="100">-->
```
Comment (jurampark) [inline src/html/dashboard.html:28]: please remove comment.

File: src/html/merge.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+</div>
+
+<div id="right-side">
+  <button id="separate">SEPARATE</button>
+  <div id="tab-list">
+    <div class="tab">
+      <div class="delete"></div>
+      <div class="title"></div>
+      <div class="url"></div>
+      <!--<img class="screen" src="../../resources/icon.png" height="100">-->
```
Comment (jurampark) [inline src/html/merge.html:27]: please remove comment.

File: src/js/dashboard.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+$(function() {
+  var tabList = $('#tab-list');
+  var tabDiv = $('.tab');
```
Comment (jurampark) [inline src/js/dashboard.js:3]: please consider about using createElement.(see http://stackoverflow.com/questions/494143/creating-a-new-dom-element-from-an-html-string-using-built-in-dom-methods-or-pro)

File: src/js/dashboard.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  tabDiv.remove();
+
+  var listNameDiv = $('.list-name');
+
+  for(var i = 0; i < localStorage.length; i++){
+    var listNameClone = listNameDiv.clone();
+    listNameClone.text(localStorage.key(i));
+    $('#left-side').append(listNameClone);
+
+    listClickListener(listNameClone);
```
Comment (jurampark) [inline src/js/dashboard.js:13]: see http://stackoverflow.com/questions/11847021/jquery-bind-vs-on.

File: src/js/command_box.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+$.ajax({
+  url: 'chrome-extension://maihpbfhkfmblcnhnmobbgikkckbfhhk/src/html/command_box.html',
+  type: 'GET',
+  success: function (data) {
+    $('body').append($(data)).css('overflow-y', 'hidden');
+
+    $('#smart-tab-manager-command-box').unbind('click').click(function(event) {
```
Comment (jurampark) [inline src/js/command_box.js:7]: maybe you don't need to do unbind.

File: src/js/manager.js [hunk 6]
```diff
@@ -4,22 +4,27 @@ var CONST_TITLE = 'title';
...
   'open': [CONST_URL, CONST_SAVED],
   'close': [CONST_ALL, CONST_URL, CONST_TITLE],
-  'order': ['time', 'name'],
+  'order': [CONST_TIME, CONST_TITLE],
   'window': [CONST_ALL, CONST_URL, CONST_TITLE],
-  'save': [CONST_ALL, CONST_URL, CONST_TITLE, CONST_VIEW],
+  'save': [CONST_ALL, CONST_URL, CONST_TITLE],
   'preview': [CONST_ALL],
+  'merge': [CONST_URL, CONST_TITLE],
   'suspend': ['older',CONST_ALL, CONST_URL, CONST_TITLE] // how to indicate all tabs?
```
Comment (jurampark) [inline src/js/manager.js:19]: please make 'older' constant

File: src/js/manager.js [hunk 7]
```diff
@@ -29,17 +34,26 @@ $(function () {
     }
   });
 });
+
 /*
  * function for change option for command selection
  */
-function changeOption() {
+function changeCommand() {
```
Comment (jurampark) [inline src/js/manager.js:41]: nit: how about commandValueChangeHandler'? its name is not clear whether it's handler or not.

File: src/js/manager.js [hunk 8]
```diff
@@ -29,17 +34,26 @@ $(function () {
...
   var option = $('#option');
   option.empty();
   var command = $('#command').val();
   for (var i = 0; i < options[command].length; i++) {
     option.append(new Option('-' + options[command][i], options[command][i]));
   }
```
Comment (jurampark) [inline src/js/manager.js:47]: How about extracting L42~47? Like
function updateOptionSelect() {
       [insert L42~47 here]
}

File: src/js/manager.js [hunk 9]
```diff
@@ -361,69 +400,126 @@ function handlePreview(indexArr) {
...
+      position: { my : "right top", at: "right bottom", collision : "fit"},
+    }).on("focus", function(){
+      $(this).autocomplete("search", '');
+    });
+  } else {
+    $('#keyword').autocomplete("destroy");
+  }
+}
+
+function showKeywordBox(){
```
Comment (jurampark) [inline src/js/manager.js:513]: Please leave comments in all functions if you have time. It's a little hard to understand without comments.

File: src/js/manager.js [hunk 10]
```diff
@@ -29,17 +34,26 @@ $(function () {
...
   option.empty();
   var command = $('#command').val();
   for (var i = 0; i < options[command].length; i++) {
     option.append(new Option('-' + options[command][i], options[command][i]));
   }
+  autocomplete();
+  showKeywordBox();
 }
+
+function changeOption(){
```
Comment (jurampark) [inline src/js/manager.js:52]: ditto, please change the name to be more clear.

File: src/js/manager.js [hunk 11]
```diff
@@ -29,17 +34,26 @@ $(function () {
...
+  autocomplete();
```
Comment (jurampark) [inline src/js/manager.js:53]: In my opinion, following flow is better for user perspective. if user change the option then
1. clear command text box.
2. trigger command text change event.
What do you think?

File: src/js/manager.js [hunk 12]
```diff
@@ -73,8 +87,27 @@ function commandSubmit() {
...
 }
+
+/*
+ * function when user empty their keyword input
+ */
+function emptyKeyword(keyword){
+  if (!keyword) {
+    $('#error-message').text('input any keyword');
+    return true;
+  }else{
```
Comment (jurampark) [inline src/js/manager.js:103]: minor style suggestion.
} else {

File: src/js/preview.js [hunk 13]
```diff
@@ -36,23 +26,28 @@ $(function () {
...
 
-  // TODO
-  function captureTabScreen(tabId, screenImg) {
-    chrome.tabs.onActivated.addListener(function () {
-      chrome.tabs.captureVisibleTab(function (capturedUrl) {
-        screenImg.attr('src', capturedUrl);
-      })
-    });
+  function addNewTab(tabInfo) {
+    var newTabDiv = tabDiv.clone();
```
Comment (jurampark) [inline src/js/preview.js:30]: ditto, please consider about using createElement.(see http://stackoverflow.com/questions/494143/creating-a-new-dom-element-from-an-html-string-using-built-in-dom-methods-or-pro)

**Review by jurampark — DISMISSED** [review]
leave some comments. please take a look!

**Review by czerny92 — APPROVED** [review]
lgtm :)

File: src/html/dashboard.html (hunk 1 above)
Comment (ttochi) [inline src/html/dashboard.html]: done

**Pull request merged**